### ISSUE TITLE
macOS runners keep hanging

### ISSUE THREAD

**Issue by OpenBagTwo** [OpenBagTwo/EnderChest#110]
## Summary

In the past week, the macOS test runners have been consistently hanging on the "Configure SSH" step:

https://github.com/OpenBagTwo/EnderChest/blob/0f7b4dae60dffd444e9939b9f4969d0d4a172fbd/.github/workflows/pull_request.yml#L37-L47

## Steps to Reproduce

Open a PR

## Workaround

If I cancel and rerun the failed jobs, they'll sometimes go through

## Severity

Painful

## Desired Outcome

1. **Research** if other people are reporting this problem and have insights into its cause.
1. If a **fix** is available, apply that fix and consider switching back to an upstream provider for the action

### Alternatives
- Don't configure local SSH at all on the macOS test runners and instead rely on the Linux workers to run the `--use-local-ssh` (non-cached) variant of the tests

## Urgency

ASAP--the workaround is painful

## Notes

This issue originally mistakenly attributed the hanging to the subsequent "install rsync" step

**Comment by OpenBagTwo** [comment]
Okay, so it's definitely hanging at the "verify that SSH is actually working" step:

```
2023-11-06T16:54:08.3802390Z +----[SHA256]-----+
2023-11-06T16:54:08.3918000Z SSH configured for local passwordless authentication
2023-11-06T17:01:08.2641470Z ##[error]The operation was canceled.
2023-11-06T17:01:08.2886070Z Post job cleanup.
2023-11-06T17:01:08.4880470Z [command]/usr/local/bin/git version
```

which makes me think I might be able to fix it with timeouts and retries.

**Comment by OpenBagTwo** [comment]
The issue was DNS resolution:

```
2023-11-06T17:06:32.1855970Z ssh: Could not resolve hostname mac-1699290219374.local: nodename nor servname provided, or not known
2023-11-06T17:06:32.1878640Z ##[error]Process completed with exit code 255.
```

:facepalm: 

This should be fixed by de6af69bf6178c68f3c41fdcda025d8545080db0 which literally just changes "$HOSTNAME" to "localhost"
